src/java/org/jivesoftware/openfire/muc/spi/MultiUserChatServiceImpl.java

```diff
@@ -386,6 +398,9 @@ else if ("http://jabber.org/protocol/disco#items".equals(namespace)) {
             IQ reply = XMPPServer.getInstance().getIQDiscoItemsHandler().handleIQ(iq);
             router.route(reply);
         }
+        else if ("urn:xmpp:ping".equals(namespace)) {
+            router.route( IQ.createResultIQ(iq) );

```

Comment:
I think we shouldn't be handling these when they refer to an occupant, only to a room or service. Otherwise a user can't reliably ping another occupant.